### ISSUE TITLE
Consistent behaviour for data_object put and create

### ISSUE THREAD

**Issue by chStaiger** [irods/python-irodsclient#132]
I found that the `data_objects.put` overwrites a data object including its metadata without a warning or throwing the `OVERWRITE_WITHOUT_FORCE_FLAG` error as the `data_objects.create` function does.
This comes a bit unexpected when comparing with the icommands.

**Comment by trel** [comment]
Certainly - thanks.

What is your proposed behavior for both the `.create` and `.put` methods?

**Comment by chStaiger** [comment]
I would prefer that also the put method throws by default the `OVERWRITE_WITHOUT_FORCE_FLAG ` and that one only overwrites the object when giving the option `kw.FORCE_FLAG_KW`.

**Comment by trel** [comment]
https://github.com/irods/python-irodsclient/blob/v0.8.0/irods/manager/data_object_manager.py#L60
and
https://github.com/irods/python-irodsclient/blob/v0.8.0/irods/manager/data_object_manager.py#L79

**Comment by michael-conway** [comment]
That seems correct. FWIW Jargon uses an overwrite keyword, as well as a
callback mechanism so that a user can intercept an overwrite and say yes,
no, yes for all, no for all. This might be a more useful behavior down the
road for interactive situations.

**Comment by d-w-moore** [comment]
Agreed.  And for the "create" case, if what you want is to guarantee an object exists without overwriting, the touch API is better.

**Comment by d-w-moore** [comment]
> And for the "create" case, if what you want is to guarantee an object exists without overwriting, the touch API is better.

Looking at it, create returns a reference to the created (or, as it currently is implemented, possibly overwritten) data object, whereas the touch method returns `None`
If the returned behavior what a developer would like to duplicate, of course, they can always dynamically augment the manager class:

```
def F(self,path,**options):
  self.touch(path,**options)
  return self.get(path)
DataObjectManager.nondestructive_create = F

```
